### DISCUSSION TITLE
Does it make sense to recruit new developers now?

### DISCUSSION THREAD

**Discussion by drbacke** [Akkudoktor-EOS/EOS#64] [General]
Do you think it makes sense for me to do some advertising on Instagram / Twitter to request additional developers. 
That would have a relatively limited effect. 
I would only do more advertising on my channel at a later stage, when the code is on a slightly better footing.

**Comment by NormannK** [comment]
imo currently its to complicated to get started. I suggest the following steps in that order:
A: find someone with knowledge how to coordinate a github project who can push commits regularly.
B: continue code cleanup, improve coding style, remove unused functions and commented sections, ignoring performance
C: improve the setup process to get it running. without at least a basic running program nobody is interested to improve it.
D: implement Tests to make sure functions work as intended. 

only after D I would advertise it. 
I'm working on a python script to made C for my setup. (i ryl don't like Node red as an extra layer)

**Reply by drbacke** [reply]
The steps are very sensefull in my opinion. 
But it's hard to find someone with knowledge, if I don't advertise it ^^
Maybe advertising is the wrong word. Recruiting would be more suitable. 
Another option would be to start a crowdfunding project and get professionals for the starting problem.

**Reply by fortrieb** [reply]
To A: It can also be a group of maintainers to coordinate the project. But it should be an uneven amount of members to move the project forward and not stuck in balanced votes.

**Reply by noootch** [reply]
I found this project from your advertisement in one of your videos. I would like to do some work in the coding style. The most important thing from my point of view is adding types. Types everywhere: args, attributes etc...this will help a lot and makes the code more safe. I already saw some places where typing would have prevented errors.

**Reply by drbacke** [reply]
Yes types could help to avoid mistakes. However, the current focus is more on the unit tests and documentation. So that we are in a position to work with a larger team. But feel free to start with the types, I think it makes a lot of sense

**Reply by noootch** [reply]
I have two PRs up that include vscode setup that I'd need for development. I could also help with tests. One PR includes a setup to debug tests in vscode.

**Reply by drbacke** [reply]
I will check them next week. Maybe we need more maintainers

**Reply by sandboxcode** [reply]
Agree to all comments before. We should keep in mind that especially in the starting phase of great community work we need motivated maintainers that work in the direction of a common goal. Paying somebody to do part of the job kind of conflicts with that at the current moment (from my perspective). 
However, it makes sense to encourage others to join us. Even it is non-code related, user input is required to describe great user stories, epics, features or even bugs. :)

**Comment by drbacke** [comment]
I've added a Milestone for priorisation
